crates/nil/src/vfs.rs

```diff
@@ -168,13 +168,14 @@ enum CodeUnitsDiff {
 
 impl LineMap {
     fn normalize(mut text: String) -> (String, Self) {
+        let text_len = text.len();
         // Must be valid for `TextSize`.
-        u32::try_from(text.len()).expect("Text too long");
+        u32::try_from(text_len).expect("Text too long");
```

Comment:
`text_len` is always used as `u32` below. We can merge these lines and save the conversion result.